core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,210 @@
+	err = contractReader.Bind(ctx, boundContracts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Get current block HEAD/tip of the blockchain to start polling from
+	latestHead, err := relayer.LatestHead(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting latestHead from relayer client: %v", err)
+	}
+	height, err := strconv.ParseUint(latestHead.Height, 10, 64)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid height in latestHead from relayer client: %v", err)
+	}
+	startBlockNum := uint64(0)
+	if height > logEventConfig.LookbackBlocks {
+		startBlockNum = height - logEventConfig.LookbackBlocks
+	}
+
+	// Setup callback channel, logger and ticker to poll ContractReader
+	callbackCh := make(chan capabilities.TriggerResponse, defaultSendChannelBufferSize)
+	ticker := time.NewTicker(time.Duration(logEventConfig.PollPeriod) * time.Millisecond)
+
+	// Initialise a Log Event Trigger
+	l := &logEventTrigger{
+		ch:   callbackCh,
+		lggr: logger.Named(lggr, fmt.Sprintf("LogEventTrigger.%s", workflowID)),
+
+		reqConfig:      reqConfig,
+		contractReader: contractReader,
+		relayer:        relayer,
+		startBlockNum:  startBlockNum,
+
+		logEventConfig: logEventConfig,
+		ticker:         ticker,
+		stopChan:       make(services.StopChan),
+		done:           make(chan bool),
+	}
+	return l, callbackCh, nil
+}
+
+func (l *logEventTrigger) Start(ctx context.Context) error {
+	go l.listen()
+	return nil
+}
+
+// Start to contract events and trigger workflow runs
+func (l *logEventTrigger) listen() {
+	ctx, cancel := l.stopChan.NewCtx()
+	defer cancel()
+	defer close(l.done)
+
+	// Listen for events from lookbackPeriod
+	var logs []types.Sequence
+	var err error
+	logData := make(map[string]any)
+	cursor := ""
+	limitAndSort := query.LimitAndSort{
+		SortBy: []query.SortBy{query.NewSortByTimestamp(query.Asc)},
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			l.lggr.Infow("Closing trigger server for (waiting for waitGroup)", "ChainID", l.logEventConfig.ChainID,
+				"ContractName", l.reqConfig.ContractName,
+				"ContractAddress", l.reqConfig.ContractAddress,
+				"ContractEventName", l.reqConfig.ContractEventName)
+			return
+		case t := <-l.ticker.C:
+			l.lggr.Infow("Polling event logs from ContractReader using QueryKey at", "time", t,
+				"startBlockNum", l.startBlockNum,
+				"cursor", cursor)
+			if cursor != "" {
+				limitAndSort.Limit = query.Limit{Cursor: cursor}
+			}
+			logs, err = l.contractReader.QueryKey(
+				ctx,
+				types.BoundContract{Name: l.reqConfig.ContractName, Address: l.reqConfig.ContractAddress},
+				query.KeyFilter{
+					Key: l.reqConfig.ContractEventName,
+					Expressions: []query.Expression{
+						query.Confidence(primitives.Finalized),
+						query.Block(fmt.Sprintf("%d", l.startBlockNum), primitives.Gte),
+					},
+				},
+				limitAndSort,
+				&logData,
+			)
+			if err != nil {
+				l.lggr.Fatalw("QueryKey failure", "err", err)
+				continue
```

Comment:
Logging Fatal level will shut down the node
```suggestion
				l.lggr.Errorw("QueryKey failure", "err", err)
				continue
```

Reply:
Sure, updated this to Errorw